### PR TITLE
Remove VersionHash used to comparison in BE

### PR DESCRIPTION
**Pull request by chaoyli**
Remove VersionHash comparison in OlapScanner and unused function in Tablet

### CODE REVIEW

**Comment by chaoyli** [comment]
#2330

File: be/src/olap/snapshot_manager.cpp [hunk 1]
```diff
@@ -396,13 +396,10 @@ OLAPStatus SnapshotManager::_create_snapshot_files(
             if (request.__isset.version) {
                 if (lastest_version->end_version() < request.version
                         || (lastest_version->start_version() == lastest_version->end_version()
-                        && lastest_version->end_version() == request.version
```
Comment (imay) [inline be/src/olap/snapshot_manager.cpp]: should remove this judge

Comment (chaoyli) [reply be/src/olap/snapshot_manager.cpp]: OK

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**